lib/node_modules/@stdlib/lapack/base/dlacpy/src/dlacpy.c

```diff
@@ -0,0 +1,212 @@
+	CBLAS_INT ia;
+	CBLAS_INT ib;
+	CBLAS_INT i0;
+	CBLAS_INT i1;
+
+	CBLAS_INT strides[] = { strideA1, strideA2 };
+
+	ia = offsetA;
+	ib = offsetB;
+	if ( stdlib_ndarray_is_row_major( 2, strides ) ) {
+		for ( i1 = 0; i1 < M; i1++ ) {
+			for ( i0 = i1; i0 < N; i0++ ) {
+				B[ ib+(i0*strideB2) ] = A[ ia+(i0*strideA2) ];
+			}
+			ia += strideA1;
+			ib += strideB1;
+		}
+		return;
+	}
+
+	for ( i1 = 0; i1 < N; i1++ ) {
+		for ( i0 = 0; i0 <= stdlib_base_fast_min( i1, M-1 ); i0++ ) {
+			B[ ib+(i0*strideB1) ] = A[ ia+(i0*strideA1) ];
+		}
+		ia += strideA2;
+		ib += strideB2;
+	}
+	return;
+}
+
+/**
+* Copies all of a matrix `A` to another matrix `B`.
+*
+* @param M         number of rows in matrix `A`
+* @param N         number of columns in matrix `A`
+* @param A         input matrix
+* @param strideA1  stride of the first dimension of `A`
+* @param strideA2  stride of the second dimension of `A`
+* @param offsetA   starting index for `A`
+* @param B         output matrix
+* @param strideB1  stride of the first dimension of `B`
+* @param strideB2  stride of the second dimension of `B`
+* @param offsetB   starting index for `B`
+*/
+void API_SUFFIX(c_copy_lower)( const CBLAS_INT M, const CBLAS_INT N, const double *A, const CBLAS_INT strideA1, const CBLAS_INT strideA2, const CBLAS_INT offsetA, double *B, const CBLAS_INT strideB1, const CBLAS_INT strideB2, const CBLAS_INT offsetB ) {
+	CBLAS_INT ia;
+	CBLAS_INT ib;
+	CBLAS_INT i0;
+	CBLAS_INT i1;
+
+	CBLAS_INT strides[] = { strideA1, strideA2 };
+
+	ia = offsetA;
+	ib = offsetB;
+	if ( stdlib_ndarray_is_row_major( 2, strides ) ) {
+		for ( i1 = 0; i1 < M; i1++ ) {
+			for ( i0 = 0; i0 <= stdlib_base_fast_min( i1, N-1 ); i0++ ) {
+				B[ ib+(i0*strideB2) ] = A[ ia+(i0*strideA2) ];
+			}
+			ia += strideA1;
+			ib += strideB1;
+		}
+		return;
+	}
+	for ( i1 = 0; i1 < N; i1++ ) {
+		for ( i0 = i1; i0 < M; i0++ ) {
+			B[ ib+(i0*strideB1) ] = A[ ia+(i0*strideA1) ];
+		}
+		ia += strideA2;
+		ib += strideB2;
+	}
+	return;
+}
+
+/**
+* Copies all or part of a matrix `A` to another matrix `B`.
+*
+* @param order  storage layout of `A` and `B`
+* @param uplo   specifies whether to copy the upper or lower triangular/trapezoidal part of matrix `A`
+* @param M      number of rows in matrix `A`
+* @param N      number of columns in matrix `A`
+* @param A      input matrix
+* @param LDA    stride of the first dimension of `A` (a.k.a., leading dimension of the matrix `A`)
+* @param B      output matrix
+* @param LDB    stride of the first dimension of `B` (a.k.a., leading dimension of the matrix `B`)
+*/
+void API_SUFFIX(c_dlacpy)( const CBLAS_LAYOUT order, const CBLAS_UPLO uplo, const CBLAS_INT M, const CBLAS_INT N, const double *A, const CBLAS_INT LDA, double *B, const CBLAS_INT LDB ) {
+	CBLAS_INT sa1;
+	CBLAS_INT sa2;
+	CBLAS_INT sb1;
+	CBLAS_INT sb2;
+
+	if ( order == CblasColMajor ) {
+		sa1 = 1;
+		sa2 = LDA;
+		sb1 = 1;
+		sb2 = LDB;
+	} else { // order === 'row-major'
+		if ( LDA < N ) {
+			return;
```

Comment:
This is wrong. You should be using `xerbla`. The JS implementation raises exceptions. That follows BLAS/LAPACK. We should do the same here.

Reply:
See CBLAS.